File: recrutplus-backend/src/main/java/com/recrutplus/recrutplus/model/Interview.java
package com.recrutplus.recrutplus.model;

import com.recrutplus.recrutplus.model.enums.InterviewStatus;
import com.recrutplus.recrutplus.model.enums.InterviewType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "interview")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Interview {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime interviewDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InterviewType type;

    private String visioLink;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id")
    private Address address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InterviewStatus status;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private LocalDateTime cancelledAt;

    @ManyToOne
    @JoinColumn(name = "application_id", nullable = false)
    private Application application;
}

File: recrutplus-frontend/src/types/Interview.ts
import type { Address } from './Address';
import type { Application } from './Application';

export enum InterviewType {
  VISIO = 'VISIO',
  PRESENTIEL = 'PRESENTIEL'
}

export enum InterviewStatus {
  PLANIFIE = 'PLANIFIE',
  TERMINE = 'TERMINE',
  ANNULE = 'ANNULE'
}

export interface Interview {
  id: number;
  interviewDate: string;
  type: InterviewType;
  visioLink?: string;
  address?: Address;
  status: InterviewStatus;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
  cancelledAt?: string;
  application: Application;
}

export interface CreateInterviewDTO {
  applicationId: number;
  interviewDate: string;
  type: InterviewType;
  visioLink?: string;
  address?: Address;
  notes?: string;
}

export interface UpdateInterviewDTO {
  interviewDate?: string;
  type?: InterviewType;
  visioLink?: string;
  address?: Address;
  status?: InterviewStatus;
  notes?: string;
}